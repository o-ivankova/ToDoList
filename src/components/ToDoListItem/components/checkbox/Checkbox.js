import React from "react";
import PropTypes from "prop-types";
import styles from "./checkbox.module.scss";

const Checkbox = ({ completed, onCheckCompleted }) => {
  return (
    <input type="checkbox" className={styles.checkbox} checked={completed} onChange={onCheckCompleted} data-testid='checkbox'/>
  );
};

Checkbox.propTypes = {
  completed: PropTypes.bool.isRequired,
  onCheckCompleted: PropTypes.func,
};

export default Checkbox;
