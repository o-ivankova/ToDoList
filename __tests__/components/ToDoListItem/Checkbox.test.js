import React from "react";
import Checkbox from "../../../src/components/ToDoListItem/components/checkbox/Checkbox";
import { jest, expect, test, describe } from "@jest/globals";
import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

const onCheckCompletedMock = jest.fn();


describe("Checkbox tests", () => {
Date.now = jest.fn(() => 1482363367071);
    test("component is rendered", () => {
      render(
          <Checkbox completed = {false} onCheckCompleted = {onCheckCompletedMock}/>
      );
  
      expect(screen.getByTestId("checkbox")).not.toBeNull;
    });
  
    test("checked value -true- displayed correctly", () => {
      render(
          <Checkbox completed = {true} onCheckCompleted = {onCheckCompletedMock}/>
      );
  
      expect(screen.getByTestId("checkbox").checked).toBe(true);
    });

    test("checked value -false- displayed correctly", () => {
        render(
            <Checkbox completed = {false} onCheckCompleted = {onCheckCompletedMock}/>
        );
    
        expect(screen.getByTestId("checkbox").checked).toBe(false);
      });

      test("onCheckCompleted function is called", () => {
        

        render(
            <Checkbox completed = {false} onCheckCompleted = {onCheckCompletedMock}/>
        );
        fireEvent.click(screen.getByTestId("checkbox"));
    
        expect(onCheckCompletedMock).toBeCalled;
      });

      test("snapshot test", () => {
        const domTree = renderer
          .create(
            <Checkbox completed = {false} onCheckCompleted = {onCheckCompletedMock}/>
          )
          .toJSON();
        expect(domTree).toMatchSnapshot();
      });
  });