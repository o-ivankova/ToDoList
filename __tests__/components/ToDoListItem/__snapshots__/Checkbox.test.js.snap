// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox tests snapshot test 1`] = `
<input
  checked={false}
  className="checkbox"
  data-testid="checkbox"
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          SyntheticBaseEvent {
            "_reactName": "onChange",
            "_targetInst": null,
            "bubbles": true,
            "cancelable": true,
            "currentTarget": null,
            "defaultPrevented": false,
            "eventPhase": 3,
            "isDefaultPrevented": [Function],
            "isPropagationStopped": [Function],
            "isTrusted": false,
            "nativeEvent": MouseEvent {
              "isTrusted": false,
            },
            "target": <input
              class="checkbox"
              data-testid="checkbox"
              type="checkbox"
            />,
            "type": "change",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  type="checkbox"
/>
`;
